src/PowerShellEditorServices/Extensions/FileContext.cs

```diff
@@ -228,6 +228,18 @@ public void InsertText(string textToInsert, BufferRange insertRange)
         }
 
         #endregion
+
+        #region File Manipulation
+
+        /// <summary>
+        /// Saves this file.
+        /// </summary>
+        public void Save()
+        {
+            this.editorOperations.SaveFile(this.scriptFile.FilePath);
```

Comment:
If you pass `ClientFilePath` here instead of `FilePath` you should be able to avoid resolving the path when it gets to VSCode.  The `InsertText` method is a good example for both sides.

Reply:
@SeeminglyScience + @daviwil: So I tried passing in `ClientFilePath`. It prefaced it with `file://`.

This is not ideal because for `openFile` and `closeFile` here:
https://github.com/PowerShell/vscode-powershell/blob/master/src/features/ExtensionCommands.ts#L362-L409

We have VSCode already resolving the paths.

So I think we should either continue resolving the paths via VSCode, or modify `openFile` and `closeFile` to expect a `ClientFilePath` as well. That way, we are consistent.

Reply:
@tylerl0706 Ah I'd keep it as is then.  In general I think it's better to have PSES do the work, but it's probably not worth changing the others.